src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,45 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            mainDiagonal[i] = matrix[i][i];
+        }
+        return mainDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int depthIndex = matrix.length - 1;
```

Comment:
good, but let's move the declaration of this variable in for loop too
Example:
https://www.geeksforgeeks.org/loop-java-important-points/
https://stackoverflow.com/a/14752913